fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/deployment/HystrixCommandInterceptor.java

```diff
@@ -113,111 +129,122 @@ public Object interceptCommand(InvocationContext ic) throws Exception {
 
         Method method = ic.getMethod();
         ExecutionContextWithInvocationContext ctx = new ExecutionContextWithInvocationContext(ic);
-        boolean shouldRunCommand = true;
-        Object res = null;
-
         LOGGER.tracef("FT operation intercepted: %s", method);
 
         CommandMetadata metadata = commandMetadataMap.computeIfAbsent(method, CommandMetadata::new);
         RetryContext retryContext = nonFallBackEnable && metadata.operation.hasRetry() ? new RetryContext(metadata.operation.getRetry()) : null;
-        SynchronousCircuitBreaker syncCircuitBreaker = null;
+        SynchronousCircuitBreaker syncCircuitBreaker = getSynchronousCircuitBreaker(metadata);
+        Function<Supplier<Object>, SimpleCommand> commandFactory = (fallback) -> new SimpleCommand(metadata.setter, ctx, fallback, metadata.operation);
 
-        while (shouldRunCommand) {
-            shouldRunCommand = false;
+        if (metadata.operation.isAsync()) {
+            LOGGER.debugf("Queue up command for async execution: %s", metadata.operation);
+            return new AsyncFuture(CompositeCommand.createAndQueue(() -> executeCommand(commandFactory, retryContext, metadata, ctx, syncCircuitBreaker), metadata.operation));
+        } else {
+            LOGGER.debugf("Sync execution: %s]", metadata.operation);
+            return executeCommand(commandFactory, retryContext, metadata, ctx, syncCircuitBreaker);
+        }
+    }
 
-            if (nonFallBackEnable && syncCircuitBreakerEnabled && metadata.hasCircuitBreaker()) {
-                syncCircuitBreaker = getSynchronousCircuitBreaker(metadata.commandKey, metadata.operation.getCircuitBreaker());
+    private static Object executeCommand(Function<Supplier<Object>, SimpleCommand> commandFactory, RetryContext retryContext, CommandMetadata metadata,
+            ExecutionContextWithInvocationContext ctx, SynchronousCircuitBreaker syncCircuitBreaker) throws Exception {
+        while (true) {
+            if (retryContext != null) {
+                LOGGER.debugf("Executing %s with %s", metadata.operation, retryContext);
             }
             Supplier<Object> fallback = null;
-            if (retryContext == null || retryContext.isLastAttempt() || (metadata.operation.isAsync() && metadata.operation.hasRetry())) {
+            if (retryContext == null || retryContext.isLastAttempt()) {
```

Comment:
what does `retryContext == null` mean in here?

Reply:
No `@Retry` annotation used -> no retry policy.